modules/post/windows/gather/enum_unattend.rb

```diff
@@ -99,8 +95,7 @@ def save_raw(xmlpath, data)
   def get_registry_unattend_path
     # HKLM\System\Setup!UnattendFile
     begin
-      key = session.sys.registry.open_key(HKEY_LOCAL_MACHINE, 'SYSTEM')
-      fname = key.query_value('Setup!UnattendFile').data
+      fname = registry_getvaldata("HKEY_LOCAL_MACHINE\\System\\Setup", "UnattendFile")&.strip
       return fname
     rescue Rex::Post::Meterpreter::RequestError
       return ''
```

Comment:
This isn't necessary any more because it's embedded within the `meterpreter_registry_getvaldata` that is called by `registry_getvaldata`. The whole exception handling block should be safe to remove. `nil` is returned by both shell sessions as well as Meterpreter sessions when the key can't be read so as long as that's handled, things will be okay.

Reply:
Okay I'll remove it :+1: